src/ituneslibrary.cs

```diff
@@ -0,0 +1,388 @@
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+using System;
+using System.ComponentModel;
+using AppKit;
+using Foundation;
+using ObjCRuntime;
+
+namespace iTunesLibrary {
+
+	[Mac (10,14, onlyOn64: true)]
+	[BaseType (typeof(NSObject))]
+	interface ITLibAlbum {
+		[NullAllowed, Export ("title")]
+		string Title { get; }
+
+		[NullAllowed, Export ("sortTitle")]
+		string SortTitle { get; }
+
+		[Export ("compilation")]
+		bool Compilation { [Bind ("isCompilation")] get; }
+
+		[NullAllowed, Export ("artist", ArgumentSemantic.Retain)]
+		ITLibArtist Artist { get; }
+
+		[Export ("discCount")]
+		nuint DiscCount { get; }
+
+		[Export ("discNumber")]
+		nuint DiscNumber { get; }
+
+		[Export ("rating")]
+		nint Rating { get; }
+
+		[Export ("ratingComputed")]
+		bool RatingComputed { [Bind ("isRatingComputed")] get; }
+
+		[Export ("gapless")]
+		bool Gapless { [Bind ("isGapless")] get; }
+
+		[Export ("trackCount")]
+		nuint TrackCount { get; }
+
+		[NullAllowed, Export ("albumArtist")]
+		string AlbumArtist { get; }
+
+		[NullAllowed, Export ("sortAlbumArtist")]
+		string SortAlbumArtist { get; }
+
+		[Export ("persistentID", ArgumentSemantic.Retain)]
+		NSNumber PersistentID { get; }
+	}
+
+	[Mac (10,14, onlyOn64: true)]
+	[BaseType (typeof(NSObject))]
+	interface ITLibArtist
+	{
+		[NullAllowed, Export ("name")]
+		string Name { get; }
+
+		[NullAllowed, Export ("sortName")]
+		string SortName { get; }
+
+		[Export ("persistentID", ArgumentSemantic.Retain)]
+		NSNumber PersistentID { get; }
+	}
+
+	[Mac (10,14, onlyOn64: true)]
+	[BaseType (typeof(NSObject))]
+	interface ITLibArtwork
+	{
+		[NullAllowed, Export ("image", ArgumentSemantic.Retain)]
+		NSImage Image { get; }
+
+		[NullAllowed, Export ("imageData", ArgumentSemantic.Retain)]
+		NSData ImageData { get; }
+
+		[Export ("imageDataFormat", ArgumentSemantic.Assign)]
+		ITLibArtworkFormat ImageDataFormat { get; }
+	}
+
+	delegate void ITLibMediaEntityEnumerateValuesHandler (NSString property, NSObject value, out bool stop);
+
+	[Mac (10,14, onlyOn64: true)]
+	[BaseType (typeof(NSObject))]
+	interface ITLibMediaEntity
+	{
+		[Export ("persistentID", ArgumentSemantic.Retain)]
+		NSNumber PersistentID { get; }
+
+		[Export ("valueForProperty:")]
+		[return: NullAllowed]
+		NSObject GetValue (string property);
+
+		[Export ("enumerateValuesForProperties:usingBlock:")]
+		void EnumerateValues (NSSet<NSString> properties, ITLibMediaEntityEnumerateValuesHandler handler);
+
+		[Export ("enumerateValuesExceptForProperties:usingBlock:")]
+		void EnumerateValuesExcluding (NSSet<NSString> properties, ITLibMediaEntityEnumerateValuesHandler handler);
```

Comment:
IMHO `EnumerateValuesExcept` sounds just as good as `EnumerateValuesExcluding`, and it has the additional benefit of being closer to the ObjC name.